src/sentry/tasks/statistical_detectors.py

```diff
@@ -118,13 +120,92 @@ def detect_transaction_trends(
 ) -> None:
     if not options.get("statistical_detectors.enable"):
         return
-    # TODO: get the transactions_per_project from an option
-    NUM_TRANSACTIONS_PER_PROJECT = 50
+
+    regressions = filter(
+        lambda trend: trend[0] == TrendType.Regressed,
+        _detect_transaction_trends(org_ids, project_ids, start),
+    )
+    for trends in chunked(regressions, TRANSACTIONS_PER_BATCH):
+        detect_transaction_change_points.delay(
+            [(payload.project_id, payload.group) for _, payload in trends],
+            start,
+        )
+
+
+@instrumented_task(
+    name="sentry.tasks.statistical_detectors.detect_transaction_change_points",
+    queue="performance.statistical_detector",
+    max_retries=0,
+)
+def detect_transaction_change_points(
+    transactions: List[Tuple[int, str | int]], start: datetime, *args, **kwargs
+) -> None:
+    for project_id, transaction in transactions:
+        with sentry_sdk.push_scope() as scope:
+            scope.set_tag("regressed_project_id", project_id)
+            scope.set_tag("regressed_transaction", transaction)
+            scope.set_context(
+                "statistical_detectors",
+                {
+                    "timestamp": start.isoformat(),
+                },
+            )
+            sentry_sdk.capture_message("Potential Transaction Regression")
+
+
+def _detect_transaction_trends(
+    org_ids: List[int], project_ids: List[int], start: datetime
+) -> Generator[Tuple[Optional[TrendType], DetectorPayload], None, None]:
+    transactions_count = 0
+    regressed_count = 0
+    improved_count = 0
```

Comment:
I would make sure to emit these as metrics otherwise this would just run in production with very little observability into what it's doing.